java-frontend/src/main/java/org/sonar/java/se/MethodYield.java

```diff
@@ -78,7 +78,7 @@ public MethodYield(int arity, boolean varArgs) {
   @Override
   public String toString() {
     return String.format("{params: %s, result: %s (%d), exceptional: %b%s}",
-      Arrays.toString(parametersConstraints),
+      parametersConstraints.stream().flatMap(MethodYield::pmapToStream).filter(Objects::nonNull).collect(Collectors.toList()),
```

Comment:
mmh, that's not going to work. As it display all the constraints from all the domains for all the parameters, you can mix which constraint is on which parameter. The test of the `toString()` method in `MethodYieldTest` should have shown you that it's not good.